echidna/core/smear.py

```diff
@@ -109,16 +109,45 @@ class EnergySmearLY(Smear):
     """ The class which smears energy. It accepts resolution in terms of light
       yield (LY) in units of NHit per MeV.
 
+    Args:
+      poisson (bool): If True, use poisson smearing.
+
     Attributes:
       _light_yield (float): The light yield of the scintillator in NHits per
         MeV.
+      _poisson_smear (Bool): True if poisson smearing is to be applied. False
+      if gaussian smearing is to be applied.
     """
 
-    def __init__(self):
+    def __init__(self, poisson=True):
         """ Initialises the class.
         """
         super(EnergySmearLY, self).__init__("energy_light_yield")
-        self._light_yield = 200.  # NHits/MeV
+        self._poisson_smear = poisson
+        self._light_yield = 200  # Nhit/MeV
+        self._log_factorial = {}
+
+    def calc_poisson_energy(self, x, lamb):
+        """ Calculates the value of a poisson whose integral is equal to
+        one at position x with a given lambda value.
+
+        Args:
+          x : Number of events
+          lamb : Lambda of the poisson
+
+        Returns:
+          float: The value of the poisson at the given position
+        """
+        photons = int(x*self._light_yield)
+        expected = lamb*self._light_yield
+        if photons not in self._log_factorial:
+            self._log_factorial[photons] = (numpy.sum
+                                            (numpy.log
+                                             (numpy.arange(1, (photons+1)))))
+        logPois = (photons*numpy.log(expected) -
```

Comment:
Again not a requirement, but I'd prefer not to mix `lowerCamelCase` and `underscore_separated` variable names.